From 755496006818d976083bcc663a872f3285c7cf1b Mon Sep 17 00:00:00 2001
From: "hong.wang" <hong.wang@amlogic.com>
Date: Sun, 23 Apr 2023 10:50:11 +0800
Subject: [PATCH] BT: The UI cannot bind the remote control [1/1]

PD#SWPL-118242

Problem:
5.15 need user confirm.

Solution:
if JUST_WORKS no need user confirm.

Verify:
T3X

Change-Id: Ie3f5574cdf24893684fd49429d53768a83610f4d
Signed-off-by: hong.wang <hong.wang@amlogic.com>
---
 net/bluetooth/smp.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/net/bluetooth/smp.c b/net/bluetooth/smp.c
index 11f853d0500f..cd2a6004fa52 100644
--- a/net/bluetooth/smp.c
+++ b/net/bluetooth/smp.c
@@ -883,6 +883,13 @@ static int tk_request(struct l2cap_conn *conn, u8 remote_oob, u8 auth,
 	    hcon->io_capability == HCI_IO_NO_INPUT_OUTPUT)
 		smp->method = JUST_WORKS;
 
+#if IS_ENABLED(AMLOGIC_LINUX_BT_SMP)
+	/* If Just Works, Continue with Zero TK */
+	if (smp->method == JUST_WORKS) {
+		set_bit(SMP_FLAG_TK_VALID, &smp->flags);
+		return 0;
+	}
+#else
 	/* If Just Works, Continue with Zero TK and ask user-space for
 	 * confirmation */
 	if (smp->method == JUST_WORKS) {
@@ -895,6 +902,7 @@ static int tk_request(struct l2cap_conn *conn, u8 remote_oob, u8 auth,
 		set_bit(SMP_FLAG_WAIT_USER, &smp->flags);
 		return 0;
 	}
+#endif
 
 	/* If this function is used for SC -> legacy fallback we
 	 * can only recover the just-works case.
-- 
2.37.1

